google-cloud-spanner-hibernate-dialect/src/main/java/com/google/cloud/spanner/hibernate/schema/SchemaUtils.java

```diff
@@ -57,4 +84,44 @@ public static Table getTable(Class<?> entityClass, Metadata metadata) {
     throw new IllegalArgumentException(
         String.format("Could not find table for entity class %s.", entityClass.getName()));
   }
+
+  /**
+   * Verifies that the composite key for an interleaved class is a super set of
+   * the parent class primary key. Assumes all tables are being verified, so does
+   * not recurse.
+   */
+  public static boolean validateInterleaved(Class<?> potentialChild) {
+    Interleaved interleaved = potentialChild.getAnnotation(Interleaved.class);
+
+    if (null == interleaved) {
+      // not interleaved, we're good
+      return true;
+    }
+
+    // TODO - when retrieving parent ids, stop after finding all the child ids
+    Set<FieldKey> childIds = resolveIdFields(potentialChild);
+    Set<FieldKey> parentIds = resolveIdFields(interleaved.parentEntity());
+
+    // Child ids should be super set of parent ids
+    return childIds.size() > parentIds.size() && childIds.containsAll(parentIds);
+  }
+
+  /**
+   * Resolve the fields that make up the composite key
+   */
+  static Set<FieldKey> resolveIdFields(Class<?> entity) {
+    Field embeddedId = getEmbeddedId(entity);
+
```

Comment:
nit: Would you mind reversing the boolean operators to be `x != null` rather than `null != x`